Sources/Commands/Workspace.swift

```diff
@@ -269,6 +273,98 @@ public class Workspace {
         return path
     }
 
+    // FIXME: Eliminate this helper method which gets revision by loading container again.
+    func clone(specifier: RepositorySpecifier, version: Version) throws -> AbsolutePath {
+        // FIXME: We need to get the revision here, and we don't have a
+        // way to get it back out of the resolver which is very
+        // annoying. Maybe we should make an SPI on the provider for
+        // this?
+        let container = try containerProvider.getContainer(for: specifier)
+        guard let tag = container.getTag(for: version) else {
+            fatalError("Resolved version: \(version) not found for \(specifier).")
+        }
+        let revision = try container.getRevision(for: tag)
+        return try self.clone(repository: specifier, at: revision, for: version)
+    }
+
+    /// This enum represents state of an external package.
+    enum PackageState {
+        /// A new package added.
+        case added(Version)
+
+        /// The package is removed.
+        case removed
+
+        /// The package is unchanged.
+        case unchanged(Version)
+
+        /// The package is updated to a new version.
+        case updated(old: Version, new: Version)
+    }
+
+    /// Updates the current dependencies.
+    public func updateDependencies() throws {
+        let rootManifest = try loadRootManifest()
+        // Only create constraints based on root manifest for the update resolution.
+        let updateConstraints = computeRootPackageConstraints(rootManifest)
+        // Resolve the dependencies.
+        let updateResults = try resolveDependencies(constraints: updateConstraints)
+        // Get the update package states from resolved results.
+        let packageStates = computePackageStates(resolvedDependencies: updateResults.map { ($0 as RepositorySpecifier, $1 as Version) })
```

Comment:
I would call this `computePackageChanges` or something, its not computing the end state.